File: java-for-impatient/ch05/Ex58.java
package com.practice.ex58;

enum Colours {
    BLACK(false, false, false),
    WHITE(true, true, true),
    RED(true, false, false),
    BLUE(false, false, true),
    GREEN(false, true, false),
    YELLOW(true, true, false),
    CYAN(false, true, true),
    MAGENTA(true, false, true);

    private final boolean red;
    private final boolean green;
    private final boolean blue;

    private Colours(boolean red, boolean green, boolean blue) {
        this.red = red;
        this.green = green;
        this.blue = blue;
    }

    public boolean getRed() {
        return red;
    }

    public boolean getGreen() {
        return green;
    }

    public boolean getBlue() {
        return blue;
    }
}

public class Ex58 {

    public static void main(String[] args) {
        var colour = Colours.RED;
        System.out.println(colour.getRed());
    }
}

File: java-for-impatient/ch05/Ex58.scala
package com.progscala3.ex58

package colors:
  enum Color(val red: Boolean, val green: Boolean, val blue: Boolean):
    case Black extends Color(false, false, false)
    case White extends Color(true, true, true)
    case Red extends Color(true, false, false)
    case Green extends Color(false, true, false)
    case Blue extends Color(false, false, true)
    case Cyan extends Color(false, true, true)
    case Magenta extends Color(true, false, true)
    case Yellow extends Color(true, true, false)

    def getRed: Boolean = red
    def getGreen: Boolean = green
    def getBlue: Boolean = blue

@main def main(args: String*): Unit =
  import colors.*

  val c = Color.Red
  println(c.getRed)
